sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,147 @@
+import { In, IsNull } from "typeorm";
+import { saveMSFAATestInputsData } from "./msfaa-factory";
+import {
+  MSFAATestInputData,
+  MSFAA_PART_TIME_MARRIED,
+  MSFAA_PART_TIME_OTHER_COUNTRY,
+  MSFAA_PART_TIME_RELATIONSHIP_OTHER,
+} from "./msfaa-part-time-process-integration.scheduler.models";
+import { OfferingIntensity } from "@sims/sims-db";
+import {
+  createMSFAARequestContentSpyOnMock,
+  getMSFAASequenceGroupName,
+  getProcessDateFromMSFAARequestContent,
+} from "./msfaa-helper";
+import * as dayjs from "dayjs";
+
+describe(
+  describeProcessorRootTest(QueueNames.PartTimeMSFAAProcessIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: PartTimeMSFAAProcessIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(PartTimeMSFAAProcessIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Expected sequence control group name.
+      const msfaaSequenceGroupName = getMSFAASequenceGroupName(
+        OfferingIntensity.partTime,
+      );
+      // Reset MSFAA part time number.
+      await db.sequenceControl.delete({ sequenceName: msfaaSequenceGroupName });
+      // Cancel any pending MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should generate an MSFAA part-time file and update the dateRequested when there are pending MSFAA records.", async () => {
+      // Arrange
+      const msfaaInputData: MSFAATestInputData[] = [
+        MSFAA_PART_TIME_MARRIED,
+        MSFAA_PART_TIME_OTHER_COUNTRY,
+        MSFAA_PART_TIME_RELATIONSHIP_OTHER,
+      ];
+      const createdMSFAARecords = await saveMSFAATestInputsData(
+        db,
+        msfaaInputData,
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      // Spy on the MSFAA content creation to intercept the process date and time used.
+      const createMSFAARequestContentMock =
+        createMSFAARequestContentSpyOnMock(app);
+
+      // Act
+      const msfaaRequestResults = await processor.processMSFAA(job);
+
+      // Assert
+      expect(createMSFAARequestContentMock).toHaveBeenCalledTimes(1);
+      const { processDate, processDateFormatted, processTimeFormatted } =
+        getProcessDateFromMSFAARequestContent(createMSFAARequestContentMock);
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      expect(uploadedFile.remoteFilePath).toBe(
+        `MSFT-Request\\DPBC.EDU.MSFA.SENT.PT.${processDateFormatted}.001`,
+      );
+      // Assert process result.
+      expect(msfaaRequestResults).toHaveLength(1);
```

Comment:
can't we directly assert like this ? 

```ts
expect(msfaaRequestResults).toBe([
        {
          generatedFile: uploadedFile.remoteFilePath,
          uploadedRecords: msfaaInputData.length,
          offeringIntensity: OfferingIntensity.partTime,
        },
      ]);
```

Reply:
Nice catch, changed but it was required to use `toStrictEqual`.